ultravox/training/evaluation.py

```diff
@@ -156,21 +160,23 @@ def evaluate(
 
         scores = [x for x in possibly_non_scores if x is not None]
 
-        if verbose:
-            print(f"Eval for {task.dataset}:")
-            for sample, score in zip(output_samples, scores):
-                print("-" * 20)
-                print(f"Q: {sample.question}")
-                print(f"A: {sample.generated_answer}")
-                print(f"X: {sample.expected_answer} [score: {score:.2f}]")
-
         average = np.mean(scores)
         std = np.std(scores) / np.sqrt(len(scores))
         metrics[f"eval_{task.name}"] = average
         metrics[f"eval_{task.name}_std"] = std
 
-        print(
-            f"Aggregate {task.metric} score for {task.dataset}: {average:.2f} ± {std:.2f}"
-        )
+        has_audio_str = "with" if task.include_audio else "without"
+        agg_score_str = f"Aggregate {task.metric} score for {task.dataset} ({has_audio_str} audio): {average:.2f} ± {std:.2f}"
+        print(agg_score_str)
+
+        if log_dir:
+            with open(os.path.join(log_dir, f"{task.name}.json"), "w") as f:
+                f.write(f"Task info: {str(task)}\n")
```

Comment:
Full task info is superior as it gives the full picture of the current evaluation scenario.